### PR TITLE
Update spring-boot version to 3.1.2

### PR DESCRIPTION
**Pull request by DanielFran**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by DanielFran** [comment]
@mshima The root cause:
_org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'filterChain' defined in class path resource [tech/jhipster/sample/config/SecurityConfiguration.class]: Failed to instantiate [org.springframework.security.web.SecurityFilterChain]: Factory method 'filterChain' threw exception with message: This method cannot decide whether these patterns are Spring MVC patterns or not. If this endpoint is a Spring MVC endpoint, please use requestMatchers(MvcRequestMatcher); otherwise, please use requestMatchers(AntPathRequestMatcher)._

**Comment by mshima** [comment]
@DanielFran just posted https://github.com/spring-projects/spring-boot/issues/36500

**Comment by mshima** [comment]
Caused by https://github.com/spring-projects/spring-security/issues/13568

**Comment by mshima** [comment]
@DanielFran are you planning to finish the update soon?

**Comment by DanielFran** [comment]
@mshima No, I am away 2 weeks from my laptop :P

**Comment by mshima** [comment]
r2dbc-mariadb is broken.
We should disable tests for it.

**Comment by mraible** [comment]
@mshima All tests seem to be passing. Is that because you replaced the MariaDB driver with MySQL? If it works, 🚢 it!

File: generators/server/templates/src/main/java/package/config/SecurityConfiguration_imperative.java.ejs [hunk 1]
```diff
@@ -180,39 +180,39 @@ public class SecurityConfiguration {
...
-                    .requestMatchers(mvc.pattern("/app/**")).permitAll()
-                    .requestMatchers(mvc.pattern("/i18n/**")).permitAll()
-                    .requestMatchers(mvc.pattern("/content/**")).permitAll()
-                    .requestMatchers(mvc.pattern("/swagger-ui/**")).permitAll()
+                    .requestMatchers(mvc.pattern("/"), AntPathRequestMatcher.antMatcher("/*.js"), AntPathRequestMatcher.antMatcher("/*.map"), AntPathRequestMatcher.antMatcher("/*.css")).permitAll()
+                    .requestMatchers(AntPathRequestMatcher.antMatcher("/*.ico"), AntPathRequestMatcher.antMatcher("/*.png"), AntPathRequestMatcher.antMatcher("/*.svg"), AntPathRequestMatcher.antMatcher("/*.webapp")).permitAll()
+                    .requestMatchers(AntPathRequestMatcher.antMatcher("/app/**")).permitAll()
+                    .requestMatchers(AntPathRequestMatcher.antMatcher("/i18n/**")).permitAll()
+                    .requestMatchers(AntPathRequestMatcher.antMatcher("/content/**")).permitAll()
+                    .requestMatchers(AntPathRequestMatcher.antMatcher("/swagger-ui/**")).permitAll()
```
Comment (gmarziou) [inline generators/server/templates/src/main/java/package/config/SecurityConfiguration_imperative.java.ejs:186]: Personally, I used a static import to make it look nicer and more concise but it's OK as it is ;)

~~~java
import static org.springframework.security.web.util.matcher.AntPathRequestMatcher.antMatcher;

                   .requestMatchers(antMatcher("/swagger-ui/**")).permitAll()
~~~

**Comment by mshima** [comment]
From https://github.com/jzheaux/cve-2023-34035-mitigations#mitigations.
We probably should match controllers urls using mvc matcher.
Don't know about `/management` and `/v3/api-docs/`.

**Comment by mraible** [comment]
@gmarziou @mshima I don't know if I agree that we should use Ant matchers in JHipster 8, even if it makes upgrading easier. Using MVC matchers is the recommended pattern from Spring Boot and I believe we should follow it.

**Comment by mshima** [comment]
> @gmarziou @mshima I don't know if I agree that we should use Ant matchers in JHipster 8, even if it makes upgrading easier. Using MVC matchers is the recommended pattern from Spring Boot and I believe we should follow it.

I think this recommendation was before the security issue was found at the link I've passed:
```
This method cannot decide whether these patterns are Spring MVC patterns or not.
If this endpoint is a Spring MVC endpoint, please use `requestMatchers(MvcRequestMatcher)`;
otherwise, please use `requestMatchers(AntPathRequestMatcher)`.
```

My initial understanding is that every resource except `/h2-console` and `/websocket` are considered `Spring MVC endpoint`. I have doubts about the ResourceHandler and management

**Comment by mraible** [comment]
@mshima OK. Thanks for the clarification. That makes sense.

**Comment by mshima** [comment]
After going through the issue and mitigation process, my understanding is that everything that goes through a DispatcherServlet should use a mvc matcher. ResourceHandler, actuator endpoints and websocket uses the DispatcherServlet.

So we should use mvc matcher for every url except h2-console which is served by a custom servlet.

I will check if actuator uses a second DispatcherServlet instance.

**Review by mraible — APPROVED** [review]

**Comment by gmarziou** [comment]
What about images and other static resources? Aren't they served by an Undertow servlet?

**Comment by mshima** [comment]
> What about images and other static resources? Aren't they served by an Undertow servlet?

They are served through spring‘s ResourceHandler that is served using DispatcherServlet.
If you extract the jar/war you will see resources inside classes/static folder instead of root folder.
I’m not sure about websocket. After a quick search seems that stomp is served by socksHandler or something like it.

> I'm still in favor of backward compatibility and stability over time.

In this case jhipster v7 using antMatcher are wrong. The best explanation I’ve found is https://github.com/spring-projects/spring-security/issues/13568#issuecomment-1654124795.

**Comment by gmarziou** [comment]
> In this case jhipster v7 using antMatcher are wrong. The best explanation I’ve found is [spring-projects/spring-security#13568 (comment)](https://github.com/spring-projects/spring-security/issues/13568#issuecomment-1654124795).

Good explanation yes but later comments show that there are issues with other servlets like h2 console and devtools, this CVE is still open so I guess there is no conclusive answer yet.

**Comment by mshima** [comment]
> Good explanation yes but later comments show that there are issues with other servlets like h2 console

There are a other things been discussed in the issue not only the security issue and fixes.
- why it needs to be a breaking change in a patch release?
- why not deprecate `requestMatcher(string)`?

If you take a look at the first commit from this PR, most of runs succeed. That happens if only 1 servlet is registered.
The issue about h2-console is that it requires an additional servlet so `requestMatcher(string)` cannot be used.

I didn’t investigate the reason for each sample failure.

The current fix for the security issue is to use mvcMatcher for Spring MVC, and make sure an DispatcherServlet with context-path (additional or not) to use like `MvcPatternBuilder.servletPath(context-path).pattern('/**')`.
And use antMatcher for non-spring servlets like `h2-console`.

File: .github/actions/setup/action.yml [hunk 2]
```diff
@@ -67,7 +67,7 @@ inputs:
   jhipster-bom-branch:
     description: 'JHipster BOM branch'
     required: false
-    default: auto
+    default: skip_ci-spring-boot_3.1.2
```
Comment (mshima) [inline .github/actions/setup/action.yml:70]:
```suggestion
    default: auto
```

File: test-integration/jdl-samples/ms-mf-react-eureka-oauth2-mariadb-infinispan/blog-store.jdl [hunk 3]
```diff
@@ -27,7 +27,7 @@ application {
     creationTimestamp 1617901618886
     jwtSecretKey "ZjY4MTM4YjI5YzMwZjhjYjI2OTNkNTRjMWQ5Y2Q0Y2YwOWNmZTE2NzRmYzU3NTMwM2NjOTE3MTllOTM3MWRkMzcyYTljMjVmNmQ0Y2MxOTUzODc0MDhhMTlkMDIxMzI2YzQzZDM2ZDE3MmQ3NjVkODk3OTVmYzljYTQyZDNmMTQ="
     packageName com.okta.developer.gateway
-    prodDatabaseType mariadb
+    prodDatabaseType mysql
```
Comment (mshima) [inline test-integration/jdl-samples/ms-mf-react-eureka-oauth2-mariadb-infinispan/blog-store.jdl:30]:
```suggestion
    # TODO switch to mariadb
    prodDatabaseType mysql
```

File: test-integration/jdl-samples/ms-mf-react-eureka-oauth2-mariadb-infinispan/blog-store.jdl [hunk 4]
```diff
@@ -89,7 +89,7 @@ application {
     entitySuffix Entity
     jwtSecretKey "ZjY4MTM4YjI5YzMwZjhjYjI2OTNkNTRjMWQ5Y2Q0Y2YwOWNmZTE2NzRmYzU3NTMwM2NjOTE3MTllOTM3MWRkMzcyYTljMjVmNmQ0Y2MxOTUzODc0MDhhMTlkMDIxMzI2YzQzZDM2ZDE3MmQ3NjVkODk3OTVmYzljYTQyZDNmMTQ="
     packageName com.okta.developer.notification
-    prodDatabaseType mariadb
+    prodDatabaseType mysql
```
Comment (mshima) [inline test-integration/jdl-samples/ms-mf-react-eureka-oauth2-mariadb-infinispan/blog-store.jdl:92]:
```suggestion
    # TODO switch to mariadb
    prodDatabaseType mysql
```

File: generators/server/templates/src/main/java/package/config/SecurityConfiguration_imperative.java.ejs [hunk 5]
```diff
@@ -178,39 +181,38 @@ public class SecurityConfiguration {
...
 <%_ if (!skipClient) { _%>
-                    .requestMatchers("/", "/index.html", "/*.js", "/*.map", "/*.css").permitAll()
-                    .requestMatchers("/*.ico", "/*.png", "/*.svg", "/*.webapp").permitAll()
-                    .requestMatchers("/app/**").permitAll()
-                    .requestMatchers("/i18n/**").permitAll()
-                    .requestMatchers("/content/**").permitAll()
-                    .requestMatchers("/swagger-ui/**").permitAll()
+                    .requestMatchers(mvc.pattern("/index.html"), mvc.pattern("/*.js"), mvc.pattern("/*.map"), mvc.pattern("/*.css")).permitAll()
+                    .requestMatchers(mvc.pattern("/*.ico"), mvc.pattern("/*.png"), mvc.pattern("/*.svg"), mvc.pattern("/*.webapp")).permitAll()
+                    .requestMatchers(mvc.pattern("/i18n/**")).permitAll()
```
Comment (mshima) [inline generators/server/templates/src/main/java/package/config/SecurityConfiguration_imperative.java.ejs:186]:
```suggestion
                    .requestMatchers(mvc.pattern("/app/**")).permitAll()
                    .requestMatchers(mvc.pattern("/i18n/**")).permitAll()
```

File: test-integration/jdl-samples/ms-mf-react-eureka-oauth2-mariadb-infinispan/blog-store.jdl [hunk 6]
```diff
@@ -27,7 +27,8 @@ application {
     creationTimestamp 1617901618886
     jwtSecretKey "ZjY4MTM4YjI5YzMwZjhjYjI2OTNkNTRjMWQ5Y2Q0Y2YwOWNmZTE2NzRmYzU3NTMwM2NjOTE3MTllOTM3MWRkMzcyYTljMjVmNmQ0Y2MxOTUzODc0MDhhMTlkMDIxMzI2YzQzZDM2ZDE3MmQ3NjVkODk3OTVmYzljYTQyZDNmMTQ="
     packageName com.okta.developer.gateway
-    prodDatabaseType mariadb
+    # TODO switch to mariadb
```
Comment (mshima) [inline test-integration/jdl-samples/ms-mf-react-eureka-oauth2-mariadb-infinispan/blog-store.jdl:30]:
```suggestion
    // TODO switch to mariadb
```

File: test-integration/jdl-samples/ms-mf-react-eureka-oauth2-mariadb-infinispan/blog-store.jdl [hunk 7]
```diff
@@ -89,7 +90,8 @@ application {
     entitySuffix Entity
     jwtSecretKey "ZjY4MTM4YjI5YzMwZjhjYjI2OTNkNTRjMWQ5Y2Q0Y2YwOWNmZTE2NzRmYzU3NTMwM2NjOTE3MTllOTM3MWRkMzcyYTljMjVmNmQ0Y2MxOTUzODc0MDhhMTlkMDIxMzI2YzQzZDM2ZDE3MmQ3NjVkODk3OTVmYzljYTQyZDNmMTQ="
     packageName com.okta.developer.notification
-    prodDatabaseType mariadb
+    # TODO switch to mariadb
```
Comment (mshima) [inline test-integration/jdl-samples/ms-mf-react-eureka-oauth2-mariadb-infinispan/blog-store.jdl:93]:
```suggestion
    // TODO switch to mariadb
```

**Comment by DanielFran** [comment]
@pascalgrimaud bounty claimed: https://opencollective.com/generator-jhipster/expenses/171152

**Pull request merged**